routers/repo/http.go

```diff
@@ -95,6 +95,13 @@ func HTTP(ctx *context.Context) {
 	}
 
 	repo, err := models.GetRepositoryByName(repoUser.ID, reponame)
+	if models.IsErrRepoNotExist(err) && setting.Repository.Upload.AutoCreate {
+
+		// if 'CreateRepository' fails, the error are handled in the next 'if err != nil' block
+		repo, err = models.CreateRepository(repoUser, repoUser, models.CreateRepoOptions{
+			Name: reponame,
```

Comment:
What about private/public repo? I think repository should be private by default.

Reply:
should it hard-coded private or with an extra bool setting - like 'AutoCreateAsPrivate'?

Reply:
I don't think it should be Auto Create specific but probably more generic like 'NewRepositoryDefaultPrivate` and this setting should be reused to set private check box to this default when creating new repository also in UI

Reply:
good point - i make a extra pr for this tomorrow

Reply:
@lafriks Actually I think it is more secure to set it private by default. In UI you can tell what you want, but if you autocreate repository, you don't have any option. Also if you use some general variable, you know about is as admin, but not as user.

Reply:
@Morlinest yes but that does not mean that can't be configurable (as I need myself to to have such option so that I don't have remember to check that check box when creating new repository :) )